### PR TITLE
Metadata: Add an option to inherit the metadata from the parent : Closes #4515

### LINKED ISSUE #4515: Metadata: Add an option to inherit the metadata from the parents
Motivation
----------
For the collaborations using Rucio as a hierarchical namespace, it can be interesting if the files inherit the metadata of all their parents.

Modification
------------
Add an inherit option in the `get_metadata` method to propagate the metadata of the parents.

### PR DESCRIPTION
**Pull request by cserf**
Metadata: Add an option to inherit the metadata from the parent : Closes #4515

### CODE REVIEW

**Comment by cserf** [comment]
This is the same code as https://github.com/rucio/rucio/pull/4621 that was mistakenly closed without being merged. The only change is the use of factories in the test to avoid noparallel and dirty markers.

File: lib/rucio/core/did.py [hunk 1]
```diff
@@ -1303,28 +1303,97 @@ def get_metadata(scope, name, plugin='DID_COLUMN', session=None):
...
+                for entry in parent_list:
+                    if entry[-1] == (child_scope, child_name):
+                        entry.append((scope, name))
+            dids = [{'scope': did[0], 'name': did[1]} for did in parents]
+            depth += 1
+        unique_dids = [{'scope': did[0], 'name': did[1]} for did in unique_dids]
+        meta_dict = {}
+        for did in unique_dids:
+            try:
+                meta = get_metadata(did['scope'], did['name'], plugin='JSON', session=session)
```
Comment (bari12) [inline lib/rucio/core/did.py:1384]: Just a note, here we need to remove the JSON force eventually.

**Comment by bari12** [comment]
I think the PR is good. @cserf any idea about the failing tests?

**Review by mlassnig — APPROVED** [review]
LGTM

once of these days we need to sort out these fixed             `for did_chunk in query.yield_per(5):` things... these hardcoded bulk values are not good

**Pull request merged**